src/main/groovy/1addJarsToClasspath.groovy

```diff
@@ -13,46 +13,59 @@ import jenkins.model.*
 
 Logger logger = Logger.getLogger("")
 
-def groovyClassLoader = this.class.classLoader
-def rootClassLoader = DefaultGroovyMethods.getRootLoader(groovyClassLoader)
-
-if (rootClassLoader == null) {
-    rootClassLoader = groovyClassLoader
-    ClassLoader parentClassLoader = rootClassLoader.getParent()
-    // climb the class hierarchy until you reach the java root loader
-    while (parentClassLoader != null) {
-        rootClassLoader = parentClassLoader
-        parentClassLoader = parentClassLoader.getParent()
-    }
+// If the workflow-cps-global-lib plugin is installed, skip the rest of this script.
+// The plugin in question manages configuring the ivy/grape system on the classpath,
+// which duplicates this script, but since the plugin is loaded before the groovy
+// scripts are initialized, will cause conflicts.
+boolean cpsLibInstalled = Jenkins.instance.getPluginManager().getPlugins().any { plugin ->
+    plugin.toString() == 'Plugin:workflow-cps-global-lib'
 }
+if (cpsLibInstalled) {
+    logger.info('The workflow-cps-global-lib plugin is installed.')
+    logger.info('No need to add ivy into the classpath again')
+    logger.info('Carrying on to the next init script')
+} else {
```

Comment:
Hmm... I know that we currently only use this script for adding ivy to the classpath, but since it supports adding multiple jars, it feels weird to escape the whole script when this plugin exists. Perhaps this logic should instead go towards L53 when the filename is ivy? Also... is there any other way of checking if ivy has already been added?